src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py

```diff
@@ -102,49 +129,105 @@ async def total_power(self) -> FormulaReceiver:
             FormulaGeneratorConfig(component_ids=self._component_ids),
         )
 
-    async def current(self, component_id: int) -> FormulaReceiver3Phase:
-        """Fetch the 3-phase current for the given EV Charger id.
+    async def component_data(self, component_id: int) -> Receiver[EVChargerData]:
+        """Stream 3-phase current values and state of an EV Charger.
 
         Args:
-            component_id: id of the EV Charger to stream current values for.
+            component_id: id of the EV Charger for which data is requested.
 
         Returns:
-            A *new* receiver that will stream 3-phase current values for the given
-                EV Charger.
-
-        Raises:
-            EVChargerPoolError: if the given component_id is not part of the pool.
+            A receiver that streams objects containing 3-phase current and state of
+                an EV Charger.
         """
-        if component_id not in self._component_ids:
-            raise EVChargerPoolError(
-                f"{component_id=} is not part of the EVChargerPool"
-                f" (with ids={self._component_ids})"
+        if recv := self._status_streams.get(component_id, None):
+            if not recv[0].done():
+                return recv[1].new_receiver()
+            logger.warning("Restarting component_status for id: %s", component_id)
+            output_chan = recv[1]
```

Comment:
I would explode the tuple here to make it easier to read, the type declaration is a bit too far away:
```suggestion
            task, output_chan = recv
            if not task.done():
                return output_chan.new_receiver()
            logger.warning("Restarting component_status for id: %s", component_id)
```